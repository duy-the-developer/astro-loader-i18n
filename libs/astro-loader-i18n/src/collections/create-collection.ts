import { i18nLoaderSchema } from "../schemas/i18n-loader-schema";
import { z } from "astro/zod";

type Options = {
  locales: string[];
  routePattern: string;
};

export function createCollection(options: Options): z.infer<typeof i18nLoaderSchema>[] {
  const { locales, routePattern } = options;
  return locales.map((locale) => ({ locale: locale, translationId: routePattern }));
}
