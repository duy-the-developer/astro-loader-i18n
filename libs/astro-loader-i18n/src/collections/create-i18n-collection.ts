import { i18nLoaderSchema } from "../schemas/i18n-loader-schema";
import { z } from "astro/zod";

type Options = {
  locales: string[];
  routePattern: string;
};

export function createI18nCollection(options: Options): { data: z.infer<typeof i18nLoaderSchema> }[] {
  const { locales, routePattern } = options;
  return locales.map((locale) => ({ data: { locale: locale, translationId: routePattern } }));
}
