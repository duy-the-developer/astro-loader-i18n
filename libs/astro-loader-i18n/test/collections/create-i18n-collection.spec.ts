import { describe, expect, it } from "vitest";
import { createI18nCollection } from "../../src/collections/create-i18n-collection";

describe("createI18nCollection", () => {
  it("should create a collection based on the provided locales and route pattern", () => {
    const locales = ["en", "de"];
    const routePattern = "/[...locale]/blog/[slug]";
    expect(createI18nCollection({ locales, routePattern })).toMatchSnapshot();
  });
});
