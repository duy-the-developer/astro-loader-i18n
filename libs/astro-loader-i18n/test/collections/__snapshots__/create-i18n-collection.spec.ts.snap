// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createI18nCollection > should create a collection based on the provided locales and route pattern 1`] = `
[
  {
    "data": {
      "locale": "en",
      "translationId": "/[...locale]/blog/[slug]",
    },
  },
  {
    "data": {
      "locale": "de",
      "translationId": "/[...locale]/blog/[slug]",
    },
  },
]
`;
